packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx

```diff
@@ -237,14 +236,14 @@ const ContractInteraction = memo(() => {
 
     // gas price
     const { value: defaultPrices } = useAsync(async () => {
-        if (networkType === NetworkType.Ethereum && !maxFeePerGas && !maxPriorityFeePerGas) {
+        if (isEIP1559Supported(chainId) && !maxFeePerGas && !maxPriorityFeePerGas) {
             const response = await WalletRPC.getEstimateGasFees(chainId)
             // Gwei to wei
             return {
                 maxPriorityFeePerGas: toHex(
-                    formatGweiToWei(response?.medium?.suggestedMaxPriorityFeePerGas ?? 0).toString(),
+                    formatGweiToWei(response?.medium?.suggestedMaxPriorityFeePerGas ?? 0).toFixed(0),
```

Comment:
It looks like you need the fixed format everywhere. Why not move toFixed into the utility function?

Reply:
This will change the output type of the utility function

Reply:
Fine,  I made a change to the utility function.